### ISSUE TITLE
[2.7.0-RC1] Unknown column 'ed.value' in 'field list'

### ISSUE THREAD

**Issue by nilshoerrmann** [symphonycms/symphony-2#2718]
Upgrading an existing site to `2.7.0-RC1` resulted in the following error in the backend:

```sql
SELECT SQL_CACHE DISTINCT `e`.`id`, `e`.section_id, `e`.`author_id`, `e`.`modification_author_id`, `e`.`creation_date` AS `creation_date`, `e`.`modification_date` AS `modification_date` , `ed`.`value` 
FROM `sym_entries` AS `e` 
WHERE 1 AND `e`.`section_id` = 3 
ORDER BY ( 
    SELECT `ed`.value 
    FROM sym_entries_data_42 AS `ed` 
    WHERE entry_id = e.id 
) 
ASC LIMIT 0, 20
```

I'm using the latest code from the `2.7.0` branch, which includes a few fixes added after the pre-release. The update script run without any errors.

**Comment by nilshoerrmann** [comment]
Similar errors are also directly printed into the front-end XML:

```xml
<data>
    <angestellte>
        <error>MySQL Error (1054): Unknown column 'ed.value' in 'field list' in query: SELECT SQL_CACHE `e`.`id`, `e`.section_id,

                `e`.`author_id`, `e`.`modification_author_id`,

                `e`.`creation_date` AS `creation_date`,

                `e`.`modification_date` AS `modification_date`

                , `ed`.`value`

            FROM `sym_entries` AS `e`

            

            WHERE 1

            AND `e`.`id` IN ('1471', '1472', '1473') 

            AND `e`.`section_id` = 7

            

            ORDER BY (

                        SELECT `ed`.value

                        FROM sym_entries_data_51 AS `ed`

                        WHERE entry_id = e.id

                    ) ASC on 964 of file /Users/xyz/Sites/example.com/symphony/lib/toolkit/class.mysql.php</error>

    </angestellte>
</data>
```

**Comment by nitriques** [comment]
Hey, this code is part of the MySQL 5.7 compat layer.

I did ran into this problem, and thought about it a lot. [The current implementation](https://github.com/symphonycms/symphony-2/blob/9668ee36581fabfd1ce8ed9b1bfcf63f76a97a9a/symphony/lib/toolkit/class.field.php#L1607-L1613) needs to stay, since fields that uses the default sort in `buildSortingSQL()` needs the default `buildSortingSelectSQL()` implementation.

The problem is this: since we wanted to have MySQL 5.7 strict support in 2.7.0, we agreed on introducing a new method for the new syntax needed. BUT, I did not thought about what would happen to fields that do provide a custom `buildSortingSQL()` implementation, but still do not implement the new `buildSortingSelectSQL()`.

When I ran into the problem, I told myself that is it extensions responsibility to add it. But again, I could pointed that this would be a breaking change.

I propose to have the current behaviour in subsequent versions, but patch it for 2.7.0 in a backward compatible way.
